### PR TITLE
fix: devcontainer script hardening and refactoring

### PR DESCRIPTION
**Pull request by marcinpsk**
- Add process-helpers.sh with graceful_kill_pid, graceful_kill_pattern, and is_expected_pid for safe process management
- Derive PLUGIN_DIR from BASH_SOURCE instead of hardcoded path
- Add CA bundle var cleanup for Compose/devcontainer empty var injection
- Rewrite netbox-stop with is_expected_pid validation and graceful kill
- Add RQ worker management (start, stop, status, logs)
- Add rq-stats, rq-jobs, rq-failed, rq-recent inspection commands
- Use uv/pip fallback in netbox-reload, plugin-install, setup.sh
- Fix superuser creation to use os.environ (avoids shell injection)
- Add robust CA cert validation (cert_count, csplit status checks)
- Enhance diagnose.sh with PLUGIN_WS_DIR derivation, empty PID check, more env info, /proc/net/tcp fallback for port check
- Enhance welcome.sh with PLUGIN_WS_DIR, Codespaces URL, GitHub CLI status
- Fix copy-paste bugs: PLUGIN_DISPLAY, devcontainer name, banner, config description all now correctly say "Data Import" instead of "Interface Name Rules"

## Summary by CodeRabbit

* **New Features**
  * New devcontainer commands and aliases for plugin lifecycle, background jobs, diagnostics, and RQ inspection.
  * Background RQ worker support and enhanced startup/stop commands.

* **Improvements**
  * Rebranded devcontainer to "Data Import" with Codespaces-aware guidance and CA/installer handling.
  * Safer process/PID handling, improved startup messages, and diagnostics.
  * Structured multi-pass import runner, centralized single-device auto-match, and safer redirect handling.
  * Import jobs now link to their associated profile view.

* **Tests**
  * Large expansion of unit and API tests covering engine, views, matching, previews, imports, and edge cases.

### CODE REVIEW

**Comment by marcinpsk** [comment]
@coderabbitai review

**Comment by marcinpsk** [comment]
@coderabbitai review

**Comment by marcinpsk** [comment]
@coderabbitai full review

**Comment by marcinpsk** [comment]
@coderabbitai full review

**Comment by marcinpsk** [comment]
@coderabbitai review

File: pyproject.toml [hunk 1]
```diff
@@ -60,18 +60,24 @@ ignore = [
...
+".devcontainer/scripts/test-e2e-preview.py" = ["C901"]
+".devcontainer/scripts/load-sample-data.py" = ["C901"]
 
 [tool.coverage.run]
 source = ["netbox_data_import"]
 omit = ["*/migrations/*", "*/tests/*"]
 
 [tool.coverage.report]
-fail_under = 80
+fail_under = 98
```
[the remark this answers is not in the document]
Comment (marcinpsk) [reply pyproject.toml:80]: we already have 98% coverage

File: pyproject.toml [hunk 2]
```diff
@@ -60,18 +60,24 @@ ignore = [
...
+max-complexity = 15
+
 [tool.ruff.lint.per-file-ignores]
 "__init__.py" = ["F401"]
 "*/migrations/*.py" = ["D"]
 "*/tests/*.py" = ["D"]
 ".devcontainer/**" = ["D"]
+".devcontainer/scripts/test-e2e.py" = ["C901"]
+".devcontainer/scripts/test-e2e-preview.py" = ["C901"]
+".devcontainer/scripts/load-sample-data.py" = ["C901"]
```
[the remark this answers is not in the document]
Comment (marcinpsk) [reply pyproject.toml:73]: they will be permanent, this is e2e test, can stay complex

**Comment by marcinpsk** [comment]
@coderabbitai review
